components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/ExportUtils.java

```diff
@@ -0,0 +1,1087 @@
+                JsonArray certificateListOfUrl = getEndpointCertificateContentAndMetaData(tenantId, url,
+                        endpointCertsDirectoryPath);
+                endpointCertificatesDetails.addAll(certificateListOfUrl);
+            }
+            if (endpointCertificatesDetails.size() > 0) {
+                writeDtoToFile(endpointCertsDirectoryPath +
+                                ImportExportConstants.ENDPOINTS_CERTIFICATE_FILE, exportFormat,
+                        ImportExportConstants.TYPE_ENDPOINT_CERTIFICATES, endpointCertificatesDetails);
+            } else if (log.isDebugEnabled()) {
+                log.debug("No endpoint certificates available for API: " + apiDto.getName() + StringUtils.SPACE
+                        + APIConstants.API_DATA_VERSION + ": " + apiDto.getVersion() + ". Skipping certificate export.");
+            }
+        } catch (JSONException e) {
+            String errorMsg = "Error in converting Endpoint config to JSON object in API: " + apiDto.getName();
+            throw new APIImportExportException(errorMsg, e);
+        } catch (IOException e) {
+            String errorMessage = "Error while retrieving saving endpoint certificate details for API: "
+                    + apiDto.getName() + " as YAML";
+            throw new APIImportExportException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Get Endpoint Certificate MetaData and Certificate detail and build JSON Array.
+     *
+     * @param tenantId          Tenant id of the user
+     * @param url               Url of the endpoint
+     * @param certDirectoryPath Directory path to export the certificates
+     * @return JSON Array of certificate details
+     * @throws APIImportExportException If an error occurs while retrieving endpoint certificate metadata and content
+     */
+    private static JsonArray getEndpointCertificateContentAndMetaData(int tenantId, String url,
+                                                                      String certDirectoryPath)
+            throws APIImportExportException {
+
+        List<CertificateMetadataDTO> certificateMetadataDTOS;
+        CertificateManager certificateManager = CertificateManagerImpl.getInstance();
+
+        try {
+            certificateMetadataDTOS = certificateManager.getCertificates(tenantId, null, url);
+        } catch (APIManagementException e) {
+            String errorMsg = "Error retrieving certificate meta data. For tenantId: " + tenantId + " hostname: "
+                    + url;
+            throw new APIImportExportException(errorMsg, e);
+        }
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        JsonArray certificatesList = new JsonArray();
+        certificateMetadataDTOS.forEach(metadataDTO -> {
+            ByteArrayInputStream certificate = null;
+            try {
+                certificate = certificateManager.getCertificateContent(metadataDTO.getAlias());
+                certificate.close();
```

Comment:
`ByteArrayInputStream certificate` is accessed after it is closed. Better to use `try whit resources` to ensure that this is not required.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9406/commits/150d9abd9d4eb25c7f1217fc53f6143f94167542